lms/djangoapps/mobile_api/video_outlines/serializers.py

```diff
@@ -190,8 +190,12 @@ def video_summary(course, course_id, video_descriptor, request, local_cache):
         video_url = video_descriptor.source
 
     # If we have the video information from VAL, we also have duration and size.
-    duration = val_video_info.get('duration', None)
-    size = val_video_info.get('file_size', 0)
+    if val_video_info.get('mobile_low'):
+        duration = val_video_info["mobile_low"].get('duration', None)
```

Comment:
I would think for `duration` and `size` we would want to do our best effort in finding the first encoding that exists - just in case there isn't a mobile_low encoding for a video.  Also, `duration` is the same for all encodings, while `size` is different for each encoding.

Given that, I recommend:
1. Sending the `size` information for each encoding separately - within the `encoded_videos` value below.  
2. For backward compatibility, the current `size` and `video_url` attributes have to remain where they are.  However, the code can be smart about finding an encoding that actually exists (just in case `mobile_low` doesn't).

You can do this by either getting the first element in the dict: 
`next(dict.itervalues())` 
or being specific about the order preference of the profiles: 
`val_video_info.get('mobile_low', val_video_info.get(`mobile_high`, val_video_info.get(`youtube`)))`